src/react/Orb.js

```diff
@@ -0,0 +1,305 @@
+  }
+
+
+ for (var k=0;k<this.newField().length;k++){
+  if (this.savedField.indexOf(this.newField()[k].__prop_addField) === -1){
+
+     var fieldIndex = columns.indexOf(this.newField()[k].__prop_addField);
+
+     if (fieldIndex !== -1){
+      this.orbFields.push({
+        name: fieldIndex.toString(),
+        caption: this.newField()[k].__prop_addField
+      });
+     }
+  
+    }
+ }
+
+    
+ for (var h=0; h<this.newField().length; h++){
+  var columnIndex = this.columnFields.indexOf(this.newField()[h] .__prop_addField);
+  var dataIndex = this.dataFields.indexOf(this.newField()[h] .__prop_addField);
+  var rowIndex = this.rowFields.indexOf(this.newField()[h] .__prop_addField);
+
+
+  if (this.newField()[h] .__prop_addField !== null){
+   switch(this.newField()[h] .__prop_location){        
+
+   case "row":
+    if (this.rowFields.indexOf(this.newField()[h] .__prop_addField) === -1){
+     this.rowFields.push(this.newField()[h] .__prop_addField);
+
+     if (columnIndex > -1){
+      this.columnFields.splice(columnIndex,1);
+     }
+     if (dataIndex > -1){
+      this.dataFields.splice(dataIndex,1);
+     }
+    }
+    
+    break;
+
+   case "column":
+    if (this.columnFields.indexOf(this.newField()[h] .__prop_addField) === -1){
+     this.columnFields.push(this.newField()[h] .__prop_addField);
+
+     if (rowIndex > -1){
+      this.rowFields.splice(columnIndex,1);
+     }
+     if (dataIndex > -1){
+      this.dataFields.splice(dataIndex,1);
+     }
+
+    }
+    break;
+
+   case "data":
+     if (this.dataFields.indexOf(this.newField()[h] .__prop_addField) === -1){
+      this.dataFields.push(this.newField()[h] .__prop_addField);
+
+      if (rowIndex > -1){
+       this.rowFields.splice(columnIndex,1);
+      }
+      if (columnIndex > -1){
+       this.columnFields.splice(dataIndex,1);
+      }
+      
+     }
+     break;
+   }
+  }
+ }
+
+
+
+ // for (var m=0; m<this.newField().length; m++){
+      
+ //  for (var n=0;n<this.orbFields.length;n++){
+    
+ //   if (this.orbFields[n].caption === this.newField()[m].__prop_addField){
+ //    var ft = this.newField()[m].__prop_formatFunction;
+
+ //    this.orbFields[n].dataSettings={
+ //     aggregateFunc:this.newField()[m].__prop_aggregateFunc,
+ //     formatFunc:function(value){
+ //      return d3.format(ft)(value);
+ //      }      
+ //    };
+
+
+ //   }
+    
+ //  }
+ // }
+
+// var context = this;
+function createFormatFunction(ft) {
+     return function (value) {
+         return d3.format(ft)(value);
+     };
+}
+for (var m = 0; m < this.newField().length; m++) {
+      
+  for (var n=0;n<this.orbFields.length;n++){
+    
+   if (this.orbFields[n].caption === this.newField()[m].__prop_addField){
+    var ft = this.newField()[m].__prop_formatFunction;
+ 
+    this.orbFields[n].dataSettings={
+     aggregateFunc:this.newField()[m].__prop_aggregateFunc,
+     formatFunc: createFormatFunction(ft)
+    };
+ 
+ 
+   }
+    
+  }
+}
+
+     
+ if (this.removeField()){
```

Comment:
Yeah. It was supposed to give dashboard designer capability to remove any field they selected first. But actually there is some bug in it---select,remove,select again---would cause some problem. I would like to remove this function now. We could add it later when we figure it out a good way for it.